src/core/components/keep-alive.js

```diff
@@ -10,10 +10,11 @@ function getComponentName (opts: ?VNodeComponentOptions): ?string {
 }
 
 function matches (pattern: string | RegExp | Array<string>, name: string): boolean {
+  name = toString(name).trim()
```

Comment:
Just wondering, do we really need to call `toString` here? Isn't `name` already a string?

Reply:
Thanks for pointing out. The name is fetched from this function:
```js
function getComponentName (opts: ?VNodeComponentOptions): ?string {
  return opts && (opts.Ctor.options.name || opts.tag)
}
```
Since the `options.name` is not required and `opt.tag` is optional: 
```js
export interface VNodeComponentOptions {
  Ctor: typeof Vue;
  propsData?: object;
  listeners?: object;
  children?: VNode[];
  tag?: string;
}
```
so here I use `toString` for double check.

Reply:
Thank you for your answer. You are right, we are getting `name` via `getComponentName`, but before calling `matches`, we are checking that `name` is not `null|undefined`:

```js
// included
(include && (!name || !matches(include, name))) ||
// excluded
(exclude && name && matches(exclude, name))
```

Reply:
What I'm trying to say is that `matches` will always be called with `name` as string. It's declared as `string` not `?string`.

- `matches` is called in `pruneCache`, there is a verification that `name` is not null.

- `matches` is called in `render`, there is a verification that `name` is not null.

Reply:
Yes, you're right, I'll update it. Thanks.